src/Umbraco.Web.BackOffice/Controllers/MemberController.cs

```diff
@@ -286,172 +308,256 @@ public MemberDisplay GetEmpty(string contentTypeAlias = null)
+                if (validatePassword.Succeeded == false)
+                {
+                    throw HttpResponseException.CreateNotificationValidationErrorResponse(validatePassword.Errors.ToErrorMessage());
+                }
 
-        private static void UpdateName(MemberSave memberSave)
-        {
-            //Don't update the name if it is empty
-            if (memberSave.Name.IsNullOrWhiteSpace() == false)
+                string newPassword = _memberManager.HashPassword(contentItem.Password.NewPassword);
+                identityMember.PasswordHash = newPassword;
+            }
+
+            IdentityResult updatedResult = await _memberManager.UpdateAsync(identityMember);
+
+            if (updatedResult.Succeeded == false)
             {
-                memberSave.PersistedContent.Name = memberSave.Name;
+                throw HttpResponseException.CreateNotificationValidationErrorResponse(updatedResult.Errors.ToErrorMessage());
             }
+
+            contentItem.PersistedContent.RawPasswordValue = identityMember.PasswordHash;
+
+            _memberService.Save(contentItem.PersistedContent);
+
+            AddOrUpdateRoles(contentItem);
         }
 
-        // TODO: This logic should be pulled into the service layer
         private async Task<bool> ValidateMemberDataAsync(MemberSave contentItem)
         {
             if (contentItem.Name.IsNullOrWhiteSpace())
             {
                 ModelState.AddPropertyError(
-                        new ValidationResult("Invalid user name", new[] { "value" }),
-                        string.Format("{0}login", Constants.PropertyEditors.InternalGenericPropertiesPrefix));
+                    new ValidationResult("Invalid user name", new[] { "value" }),
+                    $"{Constants.PropertyEditors.InternalGenericPropertiesPrefix}login");
                 return false;
             }
 
             if (contentItem.Password != null && !contentItem.Password.NewPassword.IsNullOrWhiteSpace())
             {
-                //TODO implement when NETCORE members are implemented
-                throw new NotImplementedException("TODO implement when members are implemented");
-                // var validPassword = await _passwordValidator.ValidateAsync(_passwordConfig, contentItem.Password.NewPassword);
-                // if (!validPassword)
-                // {
-                //     ModelState.AddPropertyError(
-                //        new ValidationResult("Invalid password: " + string.Join(", ", validPassword.Result), new[] { "value" }),
-                //        string.Format("{0}password", Constants.PropertyEditors.InternalGenericPropertiesPrefix));
-                //     return false;
-                // }
+                // TODO: this currently stops the user interacting with the client-side when invalid
+                IdentityResult validPassword = await _memberManager.ValidatePasswordAsync(contentItem.Password.NewPassword);
```

Comment:
Same issue as `UpdateMemberAsync` - when password is invalid, client-side message doesn't currently disappear on clicking off the field